src/store/actions.ts

```diff
@@ -152,5 +153,44 @@ export default {
     commit('updateFacetFilter', facetFilter)
     commit('updateGridSelection', gridSelection)
     commit('setToast', { type: 'success', message: 'Loaded order with id ' + id })
+  }),
+  submitOrder: tryAction(async ({ state, commit }: {state: ApplicationState, commit: any}, orderFormData: {formData: any, formFields:any }) => {
+    const { formData, formFields } = { ...orderFormData }
+    const fields = [...formFields]
+    fields.push({ id: 'contents', type: 'text' })
+    formData.contents = JSON.stringify(toCart(state))
+
+    // Generate 'unique' order number
+    formData.orderNumber = Math.floor(Math.random() * 1000000)
+    fields.push({ id: 'orderNumber', type: 'text' })
+
+    const options = {
+      headers: {
+        'Accept': 'application/json',
+        'X-Requested-With': 'XMLHttpRequest'
+      },
+      body: buildFormData(formData, fields),
+      method: 'POST',
+      credentials: 'same-origin'
+    }
+
+    let reTryCount = 0
+
+    const trySubmission = () => {
+      reTryCount++
+      options.body.set('orderNumber', Math.floor(Math.random() * 1000000).toString())
```

Comment:
repeating magic number, suggest const. Maybe a comment warning never to change it?

Reply:
A "create unique orderNumber" function would also be a solution

Reply:
done